tests/testthat/test-utilities.R

```diff
@@ -95,3 +95,31 @@ test_that("Don't split text that is marked as having originated from a control",
     #Control in Displayr can be NULL if user not selected anything yet
     expect_null(StringIsFromControl(NULL))
 })
+
+test_that("DS-4211: Names attributes are removed and data frames are returned as lists", {
+
+    num <- c(1,2,3,4,5,6,7)
+    fac <- factor(num)
+    txt <- letters[num]
+    setNames(num, txt)
+    setNames(fac, txt)
+    setNames(txt, txt)
+
+    mat <- matrix(1:12, nrow = 4)
+    rownames(mat) <- letters[1:4]
+    colnames(mat) <- letters[1:3]
+
+    df <- data.frame(fac, num, txt)
+
+    expect_null(attr(TidyDataForRVariableSet(num), "names"))
+    expect_null(attr(TidyDataForRVariableSet(fac), "names"))
+    expect_null(attr(TidyDataForRVariableSet(txt), "names"))
+    
+    tidied.df <- TidyDataForRVariableSet(df)
+    expect_false(is.data.frame(tidied.df))
+    expect_true(is.list(tidied.df))
+    expect_null(attr(tidied.df[[1]], "names"))
+    expect_null(attr(tidied.df[[2]], "names"))
+    expect_null(attr(tidied.df[[3]], "names"))
```

Comment:
A data.frame doesn't pass the row names to individual columns when subscripted.
e.g.
```
r$> df
  A B
1 1 1
2 2 2
3 3 3
4 4 4
5 5 5

r$> is.data.frame(df)
[1] TRUE

r$> names(df[[1]])
NULL

r$> rownames(df)
[1] "1" "2" "3" "4" "5"
```

Maybe update this test?